Userland/Libraries/LibTLS/Socket.cpp

```diff
@@ -205,7 +205,7 @@ bool TLSv12::check_connection_state(bool read)
                 m_context.tls_buffer.size(),
                 m_context.application_buffer.size());
         } else {
-            m_context.connection_finished = false;
+            m_context.connection_finished = true;
```

Comment:
Sorry for being misleading with this last night, setting finished to true will cause this whole chain to be retried.
Not sure if this is the cause of the regression though

Reply:
@alimpfard No worries -- I have no idea how any of this works myself, I just made changes that _seemed_ to fix my issue (`0078879`), and that _seemed_ like mistakes and made sense with the API in mind (`eb3d5ed`).

For reference, I have no interest on actually fixing this (the regression), but #8211 is for sure still valid, so if anyone has any other ideas then I'm more than welcome to implement them in this commit or let someone else take over in their own PR.